client/core-client/src/lib.rs

```diff
@@ -1,57 +1,25 @@
-// person obtaining a copy of this software and associated
-// documentation files (the "Software"), to deal in the
-// Software without restriction, including without
-// limitation the rights to use, copy, modify, merge,
-// publish, distribute, sublicense, and/or sell copies of
-// the Software, and to permit persons to whom the Software
-// is furnished to do so, subject to the following
-// conditions:
-//
-// The above copyright notice and this permission notice
-// shall be included in all copies or substantial portions
-// of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
-// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
-// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
-// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
-// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
-// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-// DEALINGS IN THE SOFTWARE.
-
-use std::convert::TryInto;
+mod helpers;
+mod manager;
+
 use std::time::Duration;
 
+use futures_channel::{mpsc, oneshot};
+use futures_util::{future::Either, SinkExt, StreamExt};
+use jsonrpsee_core::{async_trait, Error, DeserializeOwned};
+use jsonrpsee_core::client::{
+	BatchMessage, RegisterNotificationMessage, RequestMessage, Subscription, SubscriptionKind,
+	SubscriptionMessage, RequestIdManager, Client as ClientT, SubscriptionClient as SubscriptionClientT, TransportReceiver, TransportSender,
+	FrontToBack
+};
+use jsonrpsee_types::{Id, Notification, NotificationSer, RequestSer, Response, SubscriptionResponse, ParamsSer, ErrorResponse};
+use manager::RequestManager;
+use tokio::sync::Mutex;
 use crate::helpers::{
 	build_unsubscribe_message, call_with_timeout, process_batch_response, process_error_response, process_notification,
 	process_single_response, process_subscription_response, stop_subscription,
 };
-use crate::manager::RequestManager;
-use crate::transport::{Receiver as WsReceiver, Sender as WsSender, WsHandshakeError, WsTransportClientBuilder};
-use crate::types::{
-	ErrorResponse, Id, Notification, NotificationSer, ParamsSer, RequestSer, Response, SubscriptionResponse,
-	TEN_MB_SIZE_BYTES,
-};
-use async_trait::async_trait;
-use futures::channel::{mpsc, oneshot};
-use futures::future::Either;
-use futures::prelude::*;
-use futures::sink::SinkExt;
-use http::uri::{InvalidUri, Uri};
-use jsonrpsee_core::client::{
-	BatchMessage, CertificateStore, Client, FrontToBack, RegisterNotificationMessage, RequestIdManager, RequestMessage,
-	Subscription, SubscriptionClient, SubscriptionKind, SubscriptionMessage,
-};
-use jsonrpsee_core::Error;
-use serde::de::DeserializeOwned;
-use tokio::sync::Mutex;
 
-pub use soketto::handshake::client::Header;
+pub use jsonrpsee_types as types;
 
 /// Wrapper over a [`oneshot::Receiver`](futures::channel::oneshot::Receiver) that reads
```

Comment:
```suggestion
/// Wrapper over a [`oneshot::Receiver`](futures_channel::oneshot::Receiver) that reads
```

…but `oneshot::Receiver` should be in scope without the link I think?